### PR TITLE
CLOUDP-183038: Use AWS ECR for docker images in envoy-serverless

### PR DESCRIPTION
**Pull request by betsybutton**
[CLOUDP-183038](https://jira.mongodb.org/browse/CLOUDP-183038)

## Changes 
- Switches to using AWS ECR to build/store docker images used in evergreen tasks, mirroring how it's done in the serverless-proxy

## Testing 
- These changes pass most tests with the envoy upgrade to v1.25. Successful patch build: https://spruce.mongodb.com/version/6487e1aea4cf476a01fa4e8c/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC
- Note that the evergreen tasks aren't passing on this PR due to version incompatibilities, but the upgrade to v1.25 will merge moments afterward
- Followup ticket files to investigate the 8 test failures: https://jira.mongodb.org/browse/CLOUDP-184237

### CODE REVIEW

File: scripts/aws_ecr.sh [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+*)
+  echo "distro not in set"
+  usage
+  exit 1
+  ;;
+esac
+
+# Verify that SCRIPT_PATH exists and is a valid file if requested.
+if ${RUN_SCRIPT} && [ ! -f "${SCRIPT_PATH}" ]; then
+  echo "script path"
```
Comment (JamesHeppenstall) [inline scripts/aws_ecr.sh:257]: Ibid

File: scripts/aws_ecr.sh [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+  usage
+  exit 1
+fi
+
+# Verify that BUILD_DIR exists and is a valid directory if requested.
+if ${MOUNT_BUILD_DIR} && [ ! -d "${BUILD_DIR}" ]; then
+  echo "build dir"
```
Comment (JamesHeppenstall) [inline scripts/aws_ecr.sh:264]: Ibid

File: scripts/aws_ecr.sh [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+  usage
+  exit 1
+fi
+
+# Verify that ATLASPROXY_DIR exists and is a valid directory if requested.
+if ${MOUNT_ATLASPROXY_DIR} && [ ! -d "${ATLASPROXY_DIR}" ]; then
+  echo "atlas proxy dir"
```
Comment (JamesHeppenstall) [inline scripts/aws_ecr.sh:271]: Ibid

File: evergreen/Dockerfile-al2 [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+RUN curl -L -o mongodbtoolchain.tar.gz -L http://mongodbtoolchain.build.10gen.cc/toolchain/$([ $(uname -m) = "aarch64" ] && echo "amazon2-arm64" || echo "amazon2")/x86_64/latest && \
+  mkdir -p /opt/mongodbtoolchain/revisions && \
+  REVISION=$(tar --list -f mongodbtoolchain.tar.gz | head -1 | sed 's/\/$//') && \
+  tar -C /opt/mongodbtoolchain/revisions -xzvf mongodbtoolchain.tar.gz $REVISION && \
+  /opt/mongodbtoolchain/revisions/$REVISION/scripts/install.sh && \
+  rm mongodbtoolchain.tar.gz
+ENV PATH="/opt/mongodbtoolchain/v4/bin:$PATH"
+ENV LD_LIBRARY_PATH="/opt/mongodbtoolchain/v4/lib"
+ENV CC="/opt/mongodbtoolchain/v4/bin/clang"
+ENV CXX="/clang++"
```
[the remark this answers is not in the document]
Comment (betsybutton) [reply evergreen/Dockerfile-al2]: Yes! Not sure how this got in here, pretty sure I just copied and pasted the file 🤔

File: scripts/aws_ecr.sh [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+    shift
+    break
+    ;;
+  esac
+  shift
+done
+
+# Verify that DISTRO exists and is valid.
+if [ -z "${DISTRO}" ]; then
+  echo "distro"
```
[the remark this answers is not in the document]
Comment (betsybutton) [reply scripts/aws_ecr.sh:240]: 🤦‍♀️

**Review by JamesHeppenstall — APPROVED** [review]
LGTM 👍

**Pull request merged**